code/modules/bayinstruments/event_manager.dm

```diff
@@ -0,0 +1,88 @@
+/datum/musical_event
+	var/sound/object
+	var/mob/subject
+	var/datum/sound_player/source
+	var/time = 0
+	var/new_volume = 100
+
+
+/datum/musical_event/New(datum/sound_player/source, mob/subject, sound/object, time, volume)
+	src.source = source
+	src.subject = subject
+	src.object = object
+	src.time = time
+	src.new_volume = volume
```

Comment:
This will compile but is generally bad practice since Byond doesn't have anything like get;set; like in C#. Better to use different names for the inputs.